src/arcaflow_plugin_sdk/atp.py

```diff
@@ -289,6 +300,14 @@ def __str__(self):
         return self.msg
 
 
+@dataclasses.dataclass
+class StepResult():
```

Comment:
Since you're not subclassing anything, the `()` is meaningless and can be removed. Just `class StepResult:` is fine.